src/MultiField/MultiAssemblers.jl

```diff
@@ -70,34 +73,70 @@ end
 """
 MultiAssembler that produces SparseMatrices from the SparseArrays package
 """
-struct MultiSparseMatrixAssembler{E} <: MultiAssembler{SparseMatrixCSC{E,Int},Vector{E}}
+struct MultiSparseMatrixAssembler{E,M} <: MultiAssembler{AbstractSparseMatrix{E,Int},Vector{E}}
   testfesps::MultiFESpace{E}
   trialfesps::MultiFESpace{E}
+
+  function MultiSparseMatrixAssembler(
+    ::Type{M},
+    testfesp::MultiFESpace{E},
+    trialfesp::MultiFESpace{E}) where {M<:AbstractSparseMatrix,E}
+    new{E,M}(testfesp,trialfesp) 
+  end
+end
+
+function MultiSparseMatrixAssembler(
+  testfesp::MultiFESpace{E},
+  trialfesp::MultiFESpace{E}) where {E}
+  MultiSparseMatrixAssembler(SparseMatrixCSC, testfesps, trialfesp)
 end
 
 function MultiSparseMatrixAssembler(
+  ::Type{M},
   testfesps::Vector{<:FESpaceWithDirichletData},
-  trialfesps::Vector{<:FESpaceWithDirichletData})
+  trialfesps::Vector{<:FESpaceWithDirichletData}) where {M<:AbstractSparseMatrix}
   V = MultiFESpace(testfesps)
   U = MultiFESpace(trialfesps)
-  MultiSparseMatrixAssembler(V,U)
+  MultiSparseMatrixAssembler(M,V,U)
+end
+
+function MultiSparseMatrixAssembler(
+  testfesps::Vector{<:FESpaceWithDirichletData},
+  trialfesps::Vector{<:FESpaceWithDirichletData})
+  MultiSparseMatrixAssembler(SparseMatrixCSC,V,U)
+end
+
+function SparseMatrixAssembler(
+  ::Type{M},
+  testfesps::Vector{<:FESpaceWithDirichletData},
+  trialfesps::Vector{<:FESpaceWithDirichletData}) where {M<:AbstractSparseMatrix}
+  MultiSparseMatrixAssembler(M,testfesps,trialfesps)
 end
 
 function SparseMatrixAssembler(
   testfesps::Vector{<:FESpaceWithDirichletData},
   trialfesps::Vector{<:FESpaceWithDirichletData})
-  MultiSparseMatrixAssembler(testfesps,trialfesps)
+  MultiSparseMatrixAssembler(SparseMatrixCSC,testfesps,trialfesps)
 end
 
 function MultiSparseMatrixAssembler(
-  testfesps::MultiFESpace{E,S}, trialfesps::MultiFESpace{E,S}) where {E,S}
+  ::Type{M},
+  testfesps::MultiFESpace{E,S}, 
+  trialfesps::MultiFESpace{E,S}) where {M<:AbstractSparseMatrix,E,S}
   @notimplementedif S != ConsequtiveMultiFieldStyle
-  MultiSparseMatrixAssembler{E}(testfesps,trialfesps)
+  MultiSparseMatrixAssembler{E,M}(M,testfesps,trialfesps)
+end
+
+function MultiSparseMatrixAssembler(
+  testfesps::MultiFESpace{E,S}, 
+  trialfesps::MultiFESpace{E,S}) where {E,S}
+  @notimplementedif S != ConsequtiveMultiFieldStyle
+  MultiSparseMatrixAssembler(SparseMatrixCSC,testfesps,trialfesps)
 end
 
 function assemble(
-  this::MultiSparseMatrixAssembler{E},
-  allvals::Vararg{Tuple{<:MultiCellVector,<:CellNumber}}) where E
+  this::MultiSparseMatrixAssembler{E,M},
+  allvals::Vararg{Tuple{<:MultiCellVector,<:CellNumber}}) where {E,M}
```

Comment:
I thing this change is not needed (as before)

Reply:
solved in 53d940e